### PR TITLE
Add random mind swaps to bluespace anomaly

### PR DESCRIPTION
**Pull request by Simyon264**
## About the PR

Adds a random chance for bluespace anomalies to swap minds. You can swap back by hugging each other. After 60 seconds, this option is gone and you are stuck like this unless you get another bluespace anomaly or an admin uses a command to swap you both back. Mindshiels protect you from this effect during small pulses, however mindshields aren't strong enough to protect you against the anomaly going supercritical.
## Why / Balance

Adds interesting gameplay and RP situations.

## Technical details

its code 👍
(i am bad at doing technical writeups)
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
https://cdn.discordapp.com/attachments/1178043252969046030/1243739599696560178/Desktop_2024.05.25_-_03.34.53.11.mp4?ex=66529244&is=665140c4&hm=89d2919c5a6c62800bffd54fba6ea891a7484781ca2f633b52a72aa76fdc8fd4&

https://canary.discord.com/channels/1178043252969046027/1178043252969046030/1243739603614306325
**Changelog**

:cl: Simyon
- add: Bluespace anomalies now have a small chance to swap your mind with someone else near you.

### CODE REVIEW

File: Content.Server/Mind/MindSwapSystem.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        targetHuggedA.TimerCancelToken?.Cancel();
+        targetHuggedA.TimerCancelToken = new CancellationTokenSource();
+
+        var targetHuggedB = EnsureComp<RecentlySwappedComponent>(b);
+        targetHuggedB.RecentlyHugged.Clear();
+        targetHuggedB.TimerCancelToken?.Cancel();
+        targetHuggedB.TimerCancelToken = new CancellationTokenSource();
+
+        Timer.Spawn(60000, () => RemComp(b, targetHuggedB), targetHuggedB.TimerCancelToken.Token);
+        Timer.Spawn(60000, () => RemComp(a, targetHuggedA), targetHuggedA.TimerCancelToken.Token);
```
Comment (Simyon264) [inline Content.Server/Mind/MindSwapSystem.cs:122]: This whole code section contains a lot of duplicated code, but I am unsure if I should trim it down. It works fine and doesn't look too bad.

**Comment by TsjipTsjip** [comment]
DNM: There's severe roleplay issues with being forced out of your own body and having someone else take over yours, especially if you didn't get any choice in the matter and you just happened to be nearby. Pending further maintainer discussion on discord.

**Comment by Necro-a** [comment]
What stops someone from just...getting mindswapped and then going cryo as the other person? Wouldn't that just kill them?

**Comment by Simyon264** [comment]
> > (i am bad at doing technical writeups)
> 
> A bad technical writeup is better than none at all. Explain roughly what's changed and, most importantly, _why_ you've done things the way you have. If you're bad at it, do it more, not less. :)

Thats fair. I will do one.

**Comment by Simyon264** [comment]
Closed per maintainer descion.

**Pull request closed without merging**